src/nnet3/nnet-chain-example.cc

```diff
@@ -421,5 +421,49 @@ void ShiftChainExampleTimes(int32 frame_shift,
   }
 }
 
+
+void SelectFeatureOffset(int32 feature_offset, NnetChainExample *chain_eg) {
+  std::vector<NnetIo>::iterator iter = chain_eg->inputs.begin(),
+    end = chain_eg->inputs.end();
+  int32 ivec_dim, num_offsets;
```

Comment:
You can declare this ivec_dim where its used and not outside. Also it might have to be `ivector_dim`.